erts/doc/src/erlang.xml

```diff
@@ -1960,8 +2123,43 @@ true</pre>
       <name name="fun_to_list" arity="1" since=""/>
       <fsummary>Text representation of a fun.</fsummary>
       <desc>
-        <p>Returns a string corresponding to the text
-          representation of <c><anno>Fun</anno></c>.</p>
+        <warning>
+          <p>This function is <em>not</em> guaranteed to work as described here
+          when <c><anno>Fun</anno></c> was created while executing
+          uncompiled code (uncompiled escripts, the Erlang shell, and
+          other code executed by the <seeerl
+          marker="stdlib:erl_eval"><c>erl_eval</c> module</seeerl>).</p>
+        </warning>
```

Comment:
I find this warning confusing and unclear. I think it would be much better to remove it and add an example of a fun created in the shell and short explanation. I think that example and explanation should be a the end of the description.

(Our documentation guidelines says that warnings should used be when there could be serious consequences if the warnings are ignored; notes should be used for less serious things.)

Reply:
Would it be better if the warning said something like this:

"The fun_to_list/1 function does not work in a useful way when applied to fun values created while executing fun expressions in uncompiled code (uncompiled escripts, the Erlang shell, and other code executed by the <seeerl marker="stdlib:erl_eval"><c>erl_eval</c> module</seeerl>). All funs created from fun expressions in uncompiled code with the same arity are mapped to the same list by fun_to_list/1."

I agree with you that the original text in the warning was confusing and unclear but I think a warning is appropriate here as I think users can be very confused by fun_to_list/1 if they use it in an escript or shell without being aware of what the warning says. I don't think an example is needed to demonstrate what the new warning text says.

I also agree that it is better to move the warning to the end and change the text so that it first speaks about funs created by fun expressions in compiled code.

Reply:
Fixed in 5088294